microprocessor_simulator.py

```diff
@@ -16,6 +16,12 @@ def get_opcode_key(val):
     return None
 
 
+def convert_to_hex(num, bits):
+    if not isinstance(num, int):
+        raise ValueError("Invalid number type, num must be of type int.")
+    return f'{num:0{bits}x}'
```

Comment:
def convert_to_hex(num, bits):
    if not isinstance(num, int):
        raise ValueError("Invalid number type, num must be of type int.")
    return f'{num:0{int(bits/4)}x}'